heartbeat/lvmlockd

```diff
@@ -0,0 +1,438 @@
+different design and implementations. "lvmlockd" can use two lock managers:
+dlm and sanlock. This agent only supports "dlm + lvmlockd". If dlm (or corosync)
+are already being used by other cluster software, you are advised to select
+dlm, then configure "controld" resource agent for dlm and this agent for "lvmlockd".
+Otherwise, consider sanlock for "lvmlockd" if dlm/corosync is not required.
+For more information, refer to manpage lvmlockd.8.
+</longdesc>
+<shortdesc lang="en">This agent manages the lvmlockd daemon</shortdesc>
+
+<parameters>
+<parameter name="pidfile" unique="0">
+<longdesc lang="en">pid file</longdesc>
+<shortdesc lang="en">pid file</shortdesc>
+<content type="string" default="/run/lvmlockd.pid"/>
+</parameter>
+
+<parameter name="socket_path" unique="0">
+<longdesc lang="en">Set the socket path to listen on.</longdesc>
+<shortdesc lang="en">socket path</shortdesc>
+<content type="string" default="/run/lvm/lvmlockd.socket"/>
+</parameter>
+
+<parameter name="syslog_priority" unique="0">
+<longdesc lang="en">Write log messages from this level up to syslog.</longdesc>
+<shortdesc lang="en">syslog priority</shortdesc>
+<content type="string" default="warning"/>
+</parameter>
+
+<parameter name="adopt" unique="0">
+<longdesc lang="en">
+Adopt locks from a previous instance of lvmlockd.
+</longdesc>
+<shortdesc lang="en">Adopt locks from a previous instance of lvmlockd</shortdesc>
+<content type="integer" default="1"/>
+</parameter>
+
+<parameter name="activate_vgs" unique="0">
+<longdesc lang="en">
+Whether or not to activate all shared volume groups after starting
+the lvmlockd. Note that shared volume groups will always be deactivated
+before the lvmlockd stops regardless of what this option is set to.
+</longdesc>
+<shortdesc lang="en">Activate volume groups</shortdesc>
+<content type="boolean" default="true"/>
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start"		timeout="90" />
+<action name="stop"		timeout="90" />
+<action name="monitor"		timeout="90" interval="30" depth="0" />
+<action name="meta-data"	timeout="10" />
+<action name="validate-all"	timeout="20" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+: ${OCF_RESKEY_pidfile:="/run/lvmlockd.pid"}
+: ${OCF_RESKEY_activate_vgs:="true"}
+
+LOCKD="lvmlockd"
+# 0.5s sleep each count
+TIMEOUT_COUNT=20
+
+usage() {
+	cat <<END
+usage: $0 {start|stop|monitor|validate-all|meta-data}
+END
+}
+
+get_pid()
+{
+	if [ -f ${OCF_RESKEY_pidfile} ] ; then
+		cat ${OCF_RESKEY_pidfile}
+	else
+		false
+	fi
+}
+
+daemon_is_running()
+{
+	local pid=$1
+
+	# Use /proc if it exists there
+	if [ -d /proc ] && [ -d /proc/1 ] ; then
+		[ -d /proc/"$pid" ]
+	else
+		kill -s 0 "$pid" >/dev/null 2>&1
+	fi
+}
+
+silent_status()
+{
+	local pid=$(get_pid)
+
+	if [ -n "$pid" ] ; then
+		daemon_is_running "$pid"
+	else
+		# No pid file
+		false
+	fi
+}
+
+shared_vgs()
+{
+	local vg_list=()
```

Comment:
Ah, this is the problem: Your shebang is `/bin/sh`, but this is not valid POSIX shell syntax. Please either rewrite this as POSIX code, or change the shell used in the shebang to `/bin/bash`.

Reply:
OK. I tend to use `/bin/bash`, noticed many RAs have used `bash`, hah

Reply:
Yes, that's OK :)